src/mito2/src/engine/truncate_test.rs

```diff
@@ -0,0 +1,287 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::time::Duration;
+
+use api::v1::Rows;
+use common_recordbatch::RecordBatches;
+use common_telemetry::init_default_ut_logging;
+use store_api::region_engine::RegionEngine;
+use store_api::region_request::{
+    RegionCloseRequest, RegionFlushRequest, RegionOpenRequest, RegionRequest, RegionTruncateRequest,
+};
+use store_api::storage::RegionId;
+
+use super::ScanRequest;
+use crate::config::MitoConfig;
+use crate::engine::tests::{build_rows, put_rows, rows_schema};
+use crate::test_util::{CreateRequestBuilder, TestEnv};
+
+fn has_parquet_file(sst_dir: &str) -> bool {
```

Comment:
For simplicity, I think we could assert the file num of the scanner instead of adding this method to check the sst directory.
```
let scanner = engine.scan(region_id, request).unwrap();
assert_eq!(1, scanner.num_files());
```